quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HttpPropertyMappers.java

```diff
@@ -40,9 +40,11 @@ public static PropertyMapper[] getHttpPropertyMappers() {
                         .paramLabel("path")
                         .isBuildTimeProperty(true)
                         .build(),
-                builder().from("http-port")
+                HttpPropertyMappers.<Integer> builder().from("http-port")
                         .to("quarkus.http.port")
                         .defaultValue(String.valueOf(8080))
+                        .type(Integer.class)
+                        .converter(Integer::parseInt)
```

Comment:
Perhaps infer the converter based on the type and make the config even simpler?

Reply:
I like to keep the `converter` separated here since it's explicitly handled by the `Picocli` implementation and it can be used for multiple different purposes.

Reply:
I'm not saying you should remove the `converter` method from the API but infer the converter based on the type. The code above would just use:

```
type(Integer.class)
```

Reply:
What you are proposing is partially overlapping with `picocli` functionalities.
E.g. specifying an explicit type will make the default converter redundant IIUC

Reply:
I think we are talking about the same thing then. Let me ask then: Why do we need to explicitly set the `converter`?

Reply:
it was a nice extension point to build upon validation etc., I can probably take it off from the follow-up PRs to reduce the scope of the task